### DISCUSSION TITLE
Bun as a supported Corepack manager

### DISCUSSION THREAD

**Discussion by mountainash** [oven-sh/bun#4056] [Ideas]
It would be a nice "ecosystem" feature for Bun to be supported by the [Node Corepack](https://nodejs.org/api/corepack.html) feature. Does anyone have any connections with the Node team?

https://nodejs.org/api/corepack.html#supported-package-managers

I know Bun still isn't a v1 release... but Corepack is still "experimental" as well, and I think while Bun is under rapid development (inc. the `pm` Package Manager side of Bun) - it makes it a perfect time that you want to "lock" your project to a specific known working version of Bun.

**Comment by KyleSanderson** [comment]
Was looking at trying this (we just changed to pnpm a couple months ago, and bun despite not supporting windows sounds interesting in optimizing our CI even further) - but it threw an error and now I'm stuck. 😿

**Reply by mountainash** [reply]
Great story Kyle, but it's not really related to the topic if "Corepack". I suggest you seek help elsewhere or just stay with what works for you.

**Reply by KyleSanderson** [reply]
Great. I opened an issue with the project (which is where your request should have gone), but for now this can exist for continued moral support 👋

**Reply by mountainash** [reply]
> I opened an issue with the project (which is where your request should have gone)

Could you post a link to the issue, which is not really an Issue for the Bun project? It's a "discussion" about if the Bun project should seek to be recognised as a package manager by NodeJS.

**Reply by KyleSanderson** [reply]
https://github.com/nodejs/corepack/issues/295

**Comment by vincentbernat** [comment]
In https://github.com/nodejs/corepack/issues/295, there are various mentions that bun's maintainers do not wish to be supported by Corepack (notably https://github.com/nodejs/corepack/issues/295#issuecomment-2030217324, https://github.com/nodejs/corepack/issues/295#issuecomment-2030164551). Is that true?
